lib/brains/client.py

```diff
@@ -0,0 +1,95 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.http import Http404
+from django_statsd.clients import statsd
+
+import braintree
+from braintree.configuration import Configuration
+from braintree.util.xml_util import XmlUtil
+
+
+class Http(braintree.util.http.Http):
+
+    @staticmethod
+    def raise_exception_from_status(status, message=None):
+        """
+        It would be nice if a 404 from Braintree became a 404 from Django
+        as opposed to a 500 error.
+
+        This method allows us to override the exception response.
+        """
+        statsd.incr('solitude.bango.response.{0}'.format(status))
+        if status == 404:
+            raise Http404
+        return super(Http).raise_exception_from_status(status, message=message)
+
+    def __http_do(self, http_verb, path, params=None):
+        """
+        Unfortunately because this refers to staticmethods, you have
+        to override the entire class.
+        """
+        http_strategy = self.config.http_strategy()
+        request_body = XmlUtil.xml_from_dict(params) if params else ''
+        full_path = self.config.base_merchant_path() + path
+
+        try:
+            with statsd.timer('solitude.braintree.api'):
+                status, response_body = http_strategy.http_do(
+                    http_verb, full_path, self.__headers(), request_body)
+        except Exception as e:
+            if self.config.wrap_http_exceptions:
+                http_strategy.handle_exception(e)
+            else:
+                raise e
+
+        if Http.is_error_status(status):
+            Http.raise_exception_from_status(status)
+        else:
+            if len(response_body.strip()) == 0:
+                return {}
+            else:
+                return XmlUtil.dict_from_xml(response_body)
```

Comment:
This overrides way more than we need to which will be cumbersome to maintain. I know you're dealing with very java-esque Python but it looks like ``http_strategy`` has all the hooks you need. What about something like this:

````
class Http(braintree.util.http.Http):

    def http_do(self, *args, *kw):
        with statsd.timer('solitude.braintree.api'):
            return super(Http, self).http_do(*args, **kw)

    def handle_exception(self, exc):
        if exc.status == 404:
            raise Http404
        raise

Configuration(http_strategy=Http, wrap_http_exceptions=true, ...)
````